app/src/main/java/com/nexenio/bleindoorpositioningdemo/RecordingActivity.java

```diff
@@ -0,0 +1,234 @@
+
+    private final static UUID RECORDING_UUID = UUID.fromString("61a0523a-a733-4789-ae8f-4f55fcff64f2");
+
+    private RssiMeasurements rssiMeasurements = new RssiMeasurements();
+    private List<Integer> recordedRssiValues;
+    private FilteredBeaconUpdateListener<IBeacon<IBeaconAdvertisingPacket>> recordingBeaconUpdateListener;
+
+    private SharedPreferences sharedPreferences;
+
+    private TextInputEditText distanceEditText;
+    private TextInputEditText notesEditText;
+
+    private TextInputEditText deviceIdEditText;
+    private TextInputEditText deviceModelEditText;
+    private TextInputEditText deviceManufacturerEditText;
+    private TextInputEditText deviceOsVersionEditText;
+
+    private TextInputEditText beaconNameEditText;
+    private TextInputEditText beaconModelEditText;
+    private TextInputEditText beaconManufacturerEditText;
+    private TextInputEditText beaconTransmissionPowerEditText;
+    private TextInputEditText beaconAdvertisingFrequencyEditText;
+
+    private MaterialButton recordButton;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_recording);
+
+        sharedPreferences = getPreferences(Context.MODE_PRIVATE);
+
+        initializeLayout();
+        restoreFormValues();
+        initializeBluetoothScanning();
+    }
+
+    private void initializeLayout() {
+        recordButton = findViewById(R.id.recordButton);
+
+        distanceEditText = findViewById(R.id.recordDistanceEditText);
+        notesEditText = findViewById(R.id.recordNotesEditText);
+
+        deviceIdEditText = findViewById(R.id.recordDeviceIdEditText);
+        deviceModelEditText = findViewById(R.id.recordDeviceModelEditText);
+        deviceManufacturerEditText = findViewById(R.id.recordDeviceManufacturerEditText);
+        deviceOsVersionEditText = findViewById(R.id.recordDeviceOsVersionEditText);
+
+        beaconNameEditText = findViewById(R.id.recordBeaconNameEditText);
+        beaconModelEditText = findViewById(R.id.recordBeaconModelEditText);
+        beaconManufacturerEditText = findViewById(R.id.recordBeaconManufacturerEditText);
+        beaconTransmissionPowerEditText = findViewById(R.id.recordBeaconTransmissionPowerEditText);
+        beaconAdvertisingFrequencyEditText = findViewById(R.id.recordBeaconAdvertisingFrequencyEditText);
+    }
+
+    private void initializeBluetoothScanning() {
+        Log.d(TAG, "Initializing Bluetooth scanning");
+
+        BluetoothClient.initialize(this);
+        IBeaconFilter<IBeacon<IBeaconAdvertisingPacket>> recordingBeaconFilter = new IBeaconFilter<>(RECORDING_UUID);
+        recordingBeaconUpdateListener = new FilteredBeaconUpdateListener<IBeacon<IBeaconAdvertisingPacket>>(recordingBeaconFilter) {
+            @Override
+            public void onMatchingBeaconUpdated(IBeacon<IBeaconAdvertisingPacket> beacon) {
+                onRecordingBeaconUpdated(beacon);
```

Comment:
This could lead to using multiple beacon if multiple beacons use the recording UUID. If this behavior is used for now it should be well documented.